apps/wallet/src/background/keyring/Vault.ts

```diff
@@ -44,21 +56,30 @@ export class Vault {
         if (!validateEntropy(entropy)) {
             throw new Error("Can't restore Vault, entropy is invalid.");
         }
-        const vault = new Vault(entropy);
+        const vault = new Vault(entropy, keypairs);
+        const doMigrate =
+            typeof data === 'string' || data.v !== LATEST_VAULT_VERSION;
         if (doMigrate && typeof onMigrateCallback === 'function') {
             await onMigrateCallback(vault);
         }
         return vault;
     }
 
-    constructor(entropy: Uint8Array) {
-        this.entropy = entropy;
-    }
+    constructor(
+        public readonly entropy: Uint8Array,
```

Comment:
Let's not use constructor assignment. I'll add a lint rule to warn against this too.